### DISCUSSION TITLE
CUL device on /dev/ttyAMA0

### DISCUSSION THREAD

**Discussion by Morten-WM** [wmbusmeters/wmbusmeters#1148] [Q&A]
Hi,
I have a [SCC](https://www.busware.de/tiki-index.php?page=SCC) that I cannot get to work with the basic wmbusmeters daemon on a raspberry pi - and 'activated' the SCC with the gpio-pin, so it is flashing its LED.
I had it detected in fhem(https://fhem.de/) but don't need the excessive functionality of fhem, so I'm now attempting to go with the bare wmbusmeters daemon. 
I have removed the fhem package and downloaded, compiled (configure; make; make install) wmbusmeters, and started the daemon.
Probing with auto did not find anything, and knowing the device and port, I have changed /etc/wmbusmeters.conf to contain

> loglevel=trace
> device=/dev/ttyAMA0:cul:t1,s1,c1
> donotprobe=all

and get the following output:
> (lookup) with file/hex "/dev/ttyAMA0" cul
> (serial) check if /dev/ttyAMA0 can be accessed
> (serial) tty /dev/ttyAMA0 can be accessed
> (serial) before config: /dev/ttyAMA0 speed(B38400) input() output() control(CLOCAL|CREAD|CSIZE) local() special_chars() modem(DTR|RTS|CTS)
> (serial) EVENT thread interrupted
> (serial) after config:  /dev/ttyAMA0 speed(B38400) input() output() control(CLOCAL|CREAD|CSIZE) local() special_chars() modem(DTR|RTS|CTS)
> (serialtty) opened /dev/ttyAMA0 fd 4 (detect cul)
> (serial /dev/ttyAMA0) sent "560D0A"
> (serial) received binary "5620312E36372043534D3836380D0A"
> (cul) probe response "V 1.67 CSM868<0D><0A>"
> (serial /dev/ttyAMA0) sent "560D0A"
> (serial) EVENT thread interrupted
> (serial) received binary "5620312E36372043534D3836380D0A"
> (cul) probe response "V 1.67 CSM868<0D><0A>"
> (serial /dev/ttyAMA0) sent "560D0A"
> (serial) EVENT thread interrupted
> (serial) received binary "5620312E36372043534D3836380D0A"
> (cul) probe response "V 1.67 CSM868<0D><0A>"
> (serial) EVENT thread interrupted
> (serialtty) closed /dev/ttyAMA0 (detect cul)
> (cul) are you there? no

Can anyone interpret that as to why the CUL device is not being detected?

Best regards,
Morten

**Comment by Morten-WM** [comment]
Full log-file [wmbusmeters-trace.log](https://github.com/wmbusmeters/wmbusmeters/files/13996183/wmbusmeters-trace.log)

**Comment by avandorp** [comment]
I had a CUL device once and as far as I can remember it can't scan for different modes simultaneously. I'd start with one mode and add more later. I would also remove/comment out the donotprobe line. Regarding /dev/ttyAMA0: Is that the correct device?
